### ISSUE TITLE
Mounting handler functions in v0.2.2

### ISSUE THREAD

**Issue by sjukkola** [awslabs/aws-sam-cli#170]
Mounting handler functions shows inconsistent behaviour when upgrading version from `0.2.0` to `0.2.2`. Has this something to do with http router rewrite? 

I have a project structure like following
```bash
├── src
│   ├── product.js
│   └── images.js
├── templates
│   ├── template.yaml  
```

**Problem 1** 

`CodeUri: ../src` is not working anymore. Resolving module will fail. Only workaround I have found is to move `template.yaml` to project root and use `CodeUri: src/`.

**Problem 2**

Invalid logs at startup

Running`sam local start-api --template template/template.yaml` gets me following:

v0.2.0
```
Mounting images.handler (nodejs6.10) at http://127.0.0.1:3000/images [GET]
Mounting products.handler (nodejs6.10) at http://127.0.0.1:3000/products [GET]
```

v0.2.2
```
Mounting products.handler (nodejs6.10) at http://127.0.0.1:3000/images [GET]
Mounting products.handler (nodejs6.10) at http://127.0.0.1:3000/products [GET]
```

**Comment by joshuahiggins** [comment]
I was just about to an issue regarding the logs issue when I found this ticket, which happens to also cover the relative pathing issue I opened earlier today. Closing #258 now and adding what I found below:

---

As far as I can tell from reading over the source (forgive me, I've never worked in Go), the issue resides here: https://github.com/awslabs/aws-sam-local/blob/bd1672469e85bf533ee10849773ee7c92a854411/runtime.go#L262-L290

When the API runtime is instantiated, the `Cwd` is set to the template directory. However, if the `CodeUri` is a directory path than the `Cwd` will be overwritten on every invocation of an endpoint, by appending the `CodeUri` to the end of it. As such, if you have `..` (or anything starting with `../`) in your `CodeUri`, your working directory will move up one directory level for each invocation.

As an example, let's assume a project `serverless-app` has a `templates` folder for CFN templates and a `src` folder for handlers. Our API endpoint is set to `CodeUri: ..` and `Handler: src/example.handler`. 

- When local SAM starts, `Cwd` is set on the runtime as: `/Users/user/serverless-app/templates/`
- On first invoking, `Cwd` has `CodeUri` appended to it, resulting in a new path of `/Users/user/serverless-app/templates/..`. The handler works correctly.
- On second invoking, `Cwd` has `CodeUri` appended to it again, and now the path is  `/Users/user/serverless-app/..`. Not the handlers are wrong.
- Each invoking to follow does the same.

This can be countered by setting `--docker-volume-basedir $(pwd)` at start. However I felt it was worth bringing up still, as it's my personal feeling that the `Cwd` of the runtime should not be rewritten on every invocation. The working directory should remain static.
